modules/spacecraftinstruments/rendering/renderablemodelprojection.cpp

```diff
@@ -73,131 +75,66 @@ namespace {
+        ghoul::io::ModelReader::NotifyInvisibleDropped::Yes
     );
 
-    if (dictionary.hasKey(KeyGeomModelFile)) {
-        std::string file;
-
-        if (dictionary.hasValue<std::string>(KeyGeomModelFile)) {
-            // Handle single file
-            file = absPath(dictionary.value<std::string>(KeyGeomModelFile));
-            _geometry = ghoul::io::ModelReader::ref().loadModel(
-                file,
-                ghoul::io::ModelReader::ForceRenderInvisible::No,
-                ghoul::io::ModelReader::NotifyInvisibleDropped::Yes
-            );
-        }
-        else if (dictionary.hasValue<ghoul::Dictionary>(KeyGeomModelFile)) {
-            LWARNING("Loading a model with several files is deprecated and will be "
-                "removed in a future release"
-            );
-
-            ghoul::Dictionary fileDictionary = dictionary.value<ghoul::Dictionary>(
-                KeyGeomModelFile
-            );
-            std::vector<std::unique_ptr<ghoul::modelgeometry::ModelGeometry>> geometries;
-
-            for (std::string_view k : fileDictionary.keys()) {
-                // Handle each file
-                file = absPath(fileDictionary.value<std::string>(k));
-                geometries.push_back(ghoul::io::ModelReader::ref().loadModel(
-                    file,
-                    ghoul::io::ModelReader::ForceRenderInvisible::No,
-                    ghoul::io::ModelReader::NotifyInvisibleDropped::Yes
-                ));
-            }
-
-            if (!geometries.empty()) {
-                std::unique_ptr<ghoul::modelgeometry::ModelGeometry> combinedGeometry =
-                    std::move(geometries[0]);
-
-                // Combine all models into one ModelGeometry
-                for (unsigned int i = 1; i < geometries.size(); ++i) {
-                    for (ghoul::io::ModelMesh& mesh : geometries[i]->meshes()) {
-                        combinedGeometry->meshes().push_back(
-                            std::move(mesh)
-                        );
-                    }
-
-                    for (ghoul::modelgeometry::ModelGeometry::TextureEntry& texture :
-                        geometries[i]->textureStorage())
-                    {
-                        combinedGeometry->textureStorage().push_back(
-                            std::move(texture)
-                        );
-                    }
-                }
-                _geometry = std::move(combinedGeometry);
-                _geometry->calculateBoundingRadius();
-            }
-        }
-    }
-
     addPropertySubOwner(_projectionComponent);
 
     _projectionComponent.initialize(
         identifier(),
-        dictionary.value<ghoul::Dictionary>(keyProjection)
+        p.projection
     );
 
     double boundingSphereRadius = 1.0e9;
-    if (dictionary.hasValue<double>(keyBoundingSphereRadius)) {
-        boundingSphereRadius = dictionary.value<double>(keyBoundingSphereRadius);
+    if (p.boundingSphereRadius.has_value()) {
```

Comment:
More concise version to write this:
`double boundingSphereRadius = p.boundingSphereRadius.value_or(1.0e9);`